homie/examples/light.rs

```diff
@@ -0,0 +1,39 @@
+use homie::{Datatype, HomieDevice, Node, Property};
+use rumqttc::MqttOptions;
+use std::error::Error;
+use tokio::try_join;
+
+#[tokio::main(core_threads = 2)]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
+    pretty_env_logger::init();
+
+    let mqttoptions = MqttOptions::new("homie_example", "test.mosquitto.org", 1883);
+
+    let mut builder = HomieDevice::builder("homie/example", "Homie light example", mqttoptions);
+    builder.set_update_callback(|node_id, property_id, value| async move {
+        println!("{}/{} is now {}", node_id, property_id, value);
```

Comment:
It feels a bit odd to set the callback before we create the node. I feel like it might it be more natural to make the callback registration happen as part of Node creation.

Reply:
So the question is whether the callback should be per device or per property. (I don't think per node would make sense, that would be halfway between.) Any idea which is more idiomatic in Rust?

Reply:
Discussed in person, we'll keep this per-device for now.